src/components/UrlUtilsService.js

```diff
@@ -48,13 +48,44 @@ goog.provide('ga_urlutils_service');
                   /.*kmz$/.test(url));
         };
 
-        this.proxifyUrl = function(url) {
+        // Test using a head request if the remote resource enables CORS
+        this.isCorsEnabled = function(url) {
+          var deferred = $q.defer();
+          $http.head(url, { timeout: 1500 }).then(function(response) {
+            // If the head request responded successfully it means that CORS
+            // is enabled
```

Comment:
Are you sure about that assumption? Don't we need to check the response headers?

I always assumed a correctly formed pre-flight OPTIONS request is needed. See http://stackoverflow.com/questions/12173990/how-can-you-debug-a-cors-request-with-curl (among other resources, CORS is a beast)